File: StockTradingApp_Angular/Data/Transaction.cs
using System.ComponentModel.DataAnnotations;

namespace StockTradingApp_Angular.Data
{
    public class Transaction
    {
        [Key]
        public int transaction_id { get; set; }

        public int trade_id { get; set; }

        public string stock_symbol { get; set; } = string.Empty;

        public decimal? entry_price { get; set; }

        public int? shares_bought { get; set; }

        public DateTime? entry_date { get; set; }

        public decimal? dollar_stop_loss { get; set; }

        public decimal? exit_price { get; set; }

        public int? shares_sold { get; set; }

        public DateTime? exit_date { get; set; }
    }
}

File: UI/src/Models/Transaction.ts
export interface Transaction {

  transaction_id?: number;
  //trade_id: number;
  stock_symbol: string;
  entry_price: number;
  shares_bought: number;
  entry_date: Date;
  dollar_stop_loss: number;
  //exit_price: number | null;
  //shares_sold: number | null;
  //exit_date: Date | null;
}
